### ISSUE TITLE
Feature Idea Discussion: Browser Sync Support

### ISSUE THREAD

**Issue by nolsen42** [gab-ai-inc/defiant-browser#13]
I have no idea where to submit feature ideas, so I will create a issue here.

I believe that Dissenter should have the options to sync between browsers on different OSes or devices (Laptop, Desktop, Phone, etc.) Is this feature possible for Dissenter?

There are probably some privacy & data concerns, but a lot of people like myself might find this feature very useful, as I used firefox's sync on my Windows & Linux Partitions, as well as my phones and tablets.

If this feature does get implemented, I think it should follow the route firefox's syncing work since it's the most superior in my opinion, as I find chrome's syncing lacking some features last time I used it 3 years ago.

**Comment by nolsen42** [comment]
I have no idea how much space would syncing cost to gab's servers, since you have to store thousands of people's data, but people might raise an eyebrow because privacy concerns (unless you encrypt the data and somehow prove it's encrypted and trusted), but my idea is allow people host their own custom servers, and just enter the address or IP into Dissenter and begin syncing between devices.

Not everyone have their own VPS or Linux server though.

**Comment by ScottRFrost** [comment]
Something that synced with my Gab Social account would be pretty great.  Just log in to Dissenter with your Gab Social account on every computer and everything synced (with end-to-end encryption) would be amazing!

That said, I've been using https://bitwarden.com/ to sync usernames and passwords & https://www.xbrowsersync.org to sync bookmarks.  Using these 2 extensions even lets me sync between Dissenter on desktop and Firefox mobile on my Android phone.

**Comment by freespeech4ever** [comment]
Thanks for posting some alternative options @ScottRFrost 

It would definitely be nice to have, but it's not something I can deliver in the near-term for a few reasons.  I don't like these sync services personally, I want to avoid any sort of information gathering / pinging / etc, and there's just not enough time in the world to get to everything.

**Comment by trymeouteh** [comment]
I would like this too. This is the only feature keeping me from switching from Brave. Brave has created a sync feature but I find it is buggy especially on Android and only does bookmarks as of now.

I think the Dissenter should contribute to Braves Sync project and then use the code for Dissenter to polish the Sync functionality and make it stable.

The Brave Sync feature is good on privacy. No accounts needed since it uses as 24 word key and all data is encrypted in transit and at rest.

Personally all I would like to see is bookmarks synced. Anything else will be a treat.

**Comment by nolsen42** [comment]
> I want to avoid any sort of information gathering / pinging / etc

Then a good solution would be self-hosting sync servers perhaps? That way it's very easy to make sure you're in control with your (encrypted) data, for the people who must have syncing support.

We're already hosting our own servers for gab's social network, why not host our own personal sync server?

**Comment by trymeouteh** [comment]
Brave currently has disable sync but is rewriting Sync to make it solid and have more features. I think once Brave finishes their new Sync, Gab should modify it, and add it to their Dissenter Browser.

**Comment by vovchykbratyk** [comment]
A couple hours paging through the Brave forums suggests that Brave and the BAT project is on the verge of a massive bellyflop.  So in that sense I'm happy to see a fork.  Brave's enduring lack of effective sync (bookmarks, passwords, and extensions) is a major factor contributing to this situation.  I do not pretend the solution is easy - it's not.  Even Firefox's sync is super janky and unreliable.  I think Brave was getting close to a better daisy-chained solution, but suffers from very bad (or non-existent) public relations.  Sync is not itself a poison pill - it's an incredibly useful feature for those of us who hop between multiple platforms per day... and there are a _lot_ of us.  I'd be really excited to see someone besides Google competently implement it.

**Comment by trymeouteh** [comment]
> A couple hours paging through the Brave forums suggests that Brave and the BAT project is on the verge of a massive bellyflop. So in that sense I'm happy to see a fork. Brave's enduring lack of effective sync (bookmarks, passwords, and extensions) is a major factor contributing to this situation. I do not pretend the solution is easy - it's not. Even Firefox's sync is super janky and unreliable. I think Brave was getting close to a better daisy-chained solution, but suffers from very bad (or non-existent) public relations. Sync is not itself a poison pill - it's an incredibly useful feature for those of us who hop between multiple platforms per day... and there are a _lot_ of us. I'd be really excited to see someone besides Google competently implement it.

I agree. Let Brave develop Sync and then fork Brave sync for Dissenter.

**Comment by AgentZ86** [comment]
I like sync features for function but not for privacy and security. That is because they all require you to either put in your password for onedrive,dropbox or google files etc. Or if they use their own built in extention then where are the file going and who is storing them for syncing ? China ? We never know who has the data. 

For this reason I prefer a local sync on shared location perhaps with some sort of extension that might use rsync or something. 

If a remote computer or phone needs sync they should do it when back on the local area network. 
Not convenient but just a more secure atmosphere to it imo.
